lib/node/node.js

```diff
@@ -468,3 +468,56 @@ function validate(validator, params, expectedParamName) {
 
     return param;
 }
+
+Node.prototype.computeRequirements = function(databaseService, limits, callback) {
+    // By default simply compute maximum of the inputs' number of rows.
+    // TODO: if the most common multi-input analysis is some kind of join we should use
+    // the product of the input numberOfRows instead
+    var maxRows = Math.max.apply(
+        null,
+        this.inputNodes.map(function(node) { return node.estimatedRequirements.numberOfRows || 0; })
+    );
+    if (maxRows < 0) {
+        maxRows = 0;
+    }
+    this.estimatedRequirements = {
+        numberOfRows: maxRows
+    };
+    this.limits = {
+        maximumNumberOfRows: getNodeLimit(limits, this.getType(), 'maximumNumberOfRows', 1000000)
+    };
+    return callback(null, this.requirementMessages());
+};
+
+Node.prototype.requirementMessages = function() {
+    var messages = [];
+    if (this.estimatedRequirements.numberOfRows > this.limits.maximumNumberOfRows) {
+        messages.push('too many result rows');
+    }
+    return messages;
+};
+
+Node.prototype.validateRequirements = function(callback) {
+    var messages = this.requirementMessages();
+    var err;
+    if (messages.length > 0) {
+        this.status = STATUS.FAILED;
+        this.errorMessage = messages.join('\n');
+        err = new Error(this.errorMessage);
+    }
+    callback(err);
```

Comment:
I was making that function callable in an asynchronous way in case we needed to specialise it in cases in which limits (such as quotas) need to be obtain asyncrhonously.

But since we don't need async calls now and we'll change this design anyway I'll remove the callback.